krun/platform.py

```diff
@@ -647,6 +648,49 @@ def check_preliminaries(self):
         self._check_perf_samplerate()
         self._check_tickless_kernel()
         self._check_aslr_disabled()
+        self._check_realtime_throttle_disabled()
+
+    def _check_realtime_throttle_disabled(self, first=True):
+        """Linux kernel gets pretty upset if you run a CPU intensive thread
+        under the real-time thread schedule policy. By default Linux will
+        artificially pre-empt such threads to give other things a chance to run
+        on this core. A switch will flip at runtime leaving a message in dmesg
+        when this comes into effect.
+
+        See "Limiting the CPU usage of real-time and deadline processes" here:
+        Limiting the CPU usage of real-time and deadline processes
+
+        We don't want "throttling" on the benchmarking cores.
+
+        From sched(7):
+
+        "Specifying [sched_rt_runtime_us] -1 makes the runtime the same as the
+        period; that is, no CPU time is set aside for non-real-time processes."
+        """
+
+        again = False
+        with open(LinuxPlatform.SCHED_RT_RUNTIME_US) as fh:
+            val = fh.read().strip()
+
+        if val != "-1":
+            if first:
+                debug("%s is not -1, adjusting." % LinuxPlatform.SCHED_RT_RUNTIME_US)
+
+                # Needs to happen as root
+                args = self.change_user_args() +  \
+                    ["sh", "-c",
+                     "'echo -1 > %s'" % LinuxPlatform.SCHED_RT_RUNTIME_US]
+
+                cmd = " ".join(args)
+                run_shell_cmd(cmd)
+                again = True
+            else:
+                fatal("Could not set %s to -1" %
+                      LinuxPlatform.SCHED_RT_RUNTIME_US)
+
+        # We changed something. Go around again and check it took effect.
+        if again:
```

Comment:
A recursive call for something that should be a loop is dangerous (and harder to understand).